client/src/components/AddCard/AddCard.tsx

```diff
@@ -0,0 +1,58 @@
+import Paper from '@material-ui/core/Paper';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import { FormikHelpers } from 'formik';
+import Typography from '@material-ui/core/Typography';
+import useStyles from './useStyles';
+import { setDefaultPayment } from '../../helpers/APICalls/setDefaultPayment';
+import AddCardForm from './AddCardForm/AddCardForm';
+import { useAuth } from '../../context/useAuthContext';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import editProfile from '../../helpers/APICalls/editProfile';
+
+export default function AddCard(): JSX.Element {
+  const classes = useStyles();
+  const { updateSnackBarMessage } = useSnackBar();
+  const { loggedInUser, loggedInUserProfile, fetchProfileAndUpdateContext } = useAuth();
+
+  const handleSubmit = async (
+    { paymentMethod }: { paymentMethod: string },
+    { setSubmitting }: FormikHelpers<{ paymentMethod: string }>,
+  ) => {
+    if (loggedInUser) {
+      const profile = loggedInUserProfile;
```

Comment:
```suggestion
      const profile = {...loggedInUserProfile}
```
Same as comment above about directly mutating state!

Reply:
Sure. Done.